site/app/controllers/grading/ElectronicGraderController.php

```diff
@@ -48,6 +51,101 @@ private function validateDiffViewerType(string $type) {
+             $sorted_students[$reg_sec][] = $student;
+             array_push($student_list, ['user_id' => $student->getId()]);
+             array_push($student_array, $student->getId());
+        }
+        $number_of_students = count($student_list);
+        if ($number_to_grade > $number_of_students) {
+            $all_grade_all = true;
+        }
+        if ($all_grade_all) {
+            $this->core->getOutput()->renderJSONSuccess("Invalid Number of Students Entered");
+            $final_grading_info = [];
+            for ($grader = 0; $grader < count($student_array); ++$grader) {
+                $peer_array = [];
+                for ($peer = 0; $peer < count($student_array); ++$peer) {
+                    if ($grader === $peer) {
+                        continue;
+                    }
+                    else {
+                        array_push($peer_array, $student_array[$peer]);
+                    }
+                }
+                array_push($final_grading_info, [$student_array[$grader],$peer_array]);
+            }
+            $gradeable->setRandomPeerGradersList($final_grading_info);
+            return;
+        }
+        $graded_array = $student_array;
+          /*n_array_peers : An Array of arrays that holds information on to be graded peers
+          [ [A,B,C,D,E,F], [E,F,A,B,C,D], [C,D,E,F,A,B] ]
+          A grades C and E and is graded by C and E.
+          */
+        $n_array_peers = [];
+        shuffle($student_array);
+        array_push($n_array_peers, $student_array);
+          /*final_grading_info : An Array with clear structure of grading rules for peer grading
+          [ [A,[C,E]],[B,[F,D]], ...]
+          A grades C and E, B grades F and D ..and so on!
+          */
+        $final_grading_info = [];
+        $max_offset = count($student_array);
+        $offset_array = [];
+        for ($i = 0; $i < $number_to_grade; ++$i) {
+            $random_offset = rand(1, $max_offset);
+            array_push($offset_array, $random_offset);
+        }
+        foreach ($offset_array as $element) {
+            $temp_array = $student_array;
+            for ($i = 0; $i < $element; $i++) {
+                array_push($temp_array, array_shift($temp_array));
+            }
+            array_push($n_array_peers, $temp_array);
+        }
+        for ($i = 0; $i < count($n_array_peers[0]); ++$i) {
+            $temp = [];
+            for ($j = 1; $j < count($n_array_peers); ++$j) {
+                array_push($temp, $n_array_peers[$j][$i]);
+            }
+            array_push($final_grading_info, [$n_array_peers[0][$i],$temp]);
+        }
+        $this->core->getOutput()->renderJsonSuccess($final_grading_info);
+        $gradeable->setRandomPeerGradersList($final_grading_info);
+        if ($number_to_grade < 1) {
+            $this->core->getOutput()->renderJsonError("Clear Peer Matrix");
```

Comment:
Instead of using `renderJsonSucess` and `renderJsonError` I reccomend you use the response object and expicitly return a json response which is more clear.

For example you can do
```php
return MultiResponse::JsonOnlyResponse(
    JsonResponse::getSuccessResponse($final_grading_info)
);
```
on success  and 
```php
return MultiResponse::JsonOnlyResponse(
    JsonResponse::getFailResponse("Clear Peer Matrix")
);

on failure
```

https://submitty.org/developer/development_instructions/router-response

Reply:
Updated!